packages/std/src/math/uint256.rs

```diff
@@ -0,0 +1,560 @@
+    }
+}
+
+impl ops::SubAssign<Uint256> for Uint256 {
+    fn sub_assign(&mut self, rhs: Uint256) {
+        self.0 = self.0.checked_sub(rhs.0).unwrap();
+    }
+}
+
+impl<'a> ops::SubAssign<&'a Uint256> for Uint256 {
+    fn sub_assign(&mut self, rhs: &'a Uint256) {
+        self.0 = self.0.checked_sub(rhs.0).unwrap();
+    }
+}
+
+impl ops::DivAssign<Uint256> for Uint256 {
+    fn div_assign(&mut self, rhs: Self) {
+        self.0 = self.0.checked_div(rhs.0).unwrap();
+    }
+}
+
+impl<'a> ops::DivAssign<&'a Uint256> for Uint256 {
+    fn div_assign(&mut self, rhs: &'a Uint256) {
+        self.0 = self.0.checked_div(rhs.0).unwrap();
+    }
+}
+
+impl ops::ShrAssign<u32> for Uint256 {
+    fn shr_assign(&mut self, rhs: u32) {
+        if rhs > 256 {
+            panic!(
+                "right shift error: {} is larger than the number of bits in Uint256",
+                rhs
+            );
+        }
+
+        self.0 = self.0.shr(rhs);
+    }
+}
+
+impl<'a> ops::ShrAssign<&'a u32> for Uint256 {
+    fn shr_assign(&mut self, rhs: &'a u32) {
+        if *rhs > 256 {
+            panic!(
+                "right shift error: {} is larger than the number of bits in Uint256",
+                rhs
+            );
+        }
+
+        self.0 = self.0.shr(*rhs);
+    }
+}
+
+impl Serialize for Uint256 {
+    /// Serializes as an integer string using base 10
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for Uint256 {
+    /// Deserialized from an integer string using base 10
+    fn deserialize<D>(deserializer: D) -> Result<Uint256, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(Uint256Visitor)
+    }
+}
+
+struct Uint256Visitor;
+
+impl<'de> de::Visitor<'de> for Uint256Visitor {
+    type Value = Uint256;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("string-encoded integer")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        match U256::from_dec_str(v) {
+            Ok(u) => Ok(Uint256(u)),
+            Err(e) => Err(E::custom(format!("invalid Uint256 '{}' - {}", v, e))),
+        }
+    }
+}
+
+impl Sum<Uint256> for Uint256 {
+    fn sum<I: Iterator<Item = Uint256>>(iter: I) -> Self {
+        iter.fold(Uint256::zero(), ops::Add::add)
+    }
+}
+
+impl<'a> Sum<&'a Uint256> for Uint256 {
+    fn sum<I: Iterator<Item = &'a Uint256>>(iter: I) -> Self {
+        iter.fold(Uint256::zero(), ops::Add::add)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{from_slice, to_vec};
+
+    #[test]
+    fn uint256_construct() {
+        let original = Uint256::new([1; 32]);
+        let a: [u8; 32] = original.big_endian();
+        assert_eq!(a, [1; 32]);
```

Comment:
Let's add a test with different values in the bytes to ensure endianess is right